lib/ansible/modules/system/pamd.py

```diff
@@ -545,9 +618,12 @@ def main():
             change, result = add_module_arguments(pamd,
                                                   old_rule,
                                                   module_arguments)
+        elif state == 'absent':
+            change, result = remove_rule(pamd,
+                                         old_rule)
 
-        if not module.check_mode:
-            write_rules(pamd)
+        if not module.check_mode and change:
+            pamd.write()
 
     except Exception:
```

Comment:
we discourage catchall  exceptions as they give poor feedback to the user,  use trapping closer to source to give a more accurate error and if possible give the user a hint on what they can do to resolve the issue